### ISSUE TITLE
Does not remember location of Windows apps

### ISSUE THREAD

**Issue by tomludd** [manutalcual/winredock#31]
Would be nice if Windows Mail and Calendar also was moved 👍

I use the Windows Mail and Calendar apps:
https://www.microsoft.com/store/productId/9WZDNCRFHVQM

**Comment by manutalcual** [comment]
Yes. I know. Unfortunately I don't have the time to acquire the knowledge to fix that. It seems MS does things in a different way.
I would like to learn and adapt the soft but, as said, I don't have enough time.
I'm sorry.
Anyway, thanks to use it and to take the time to notice. (And sorry to be so late to give a response.)

**Comment by stevennyman** [comment]
This may be helpful in resolving this issue: https://docs.microsoft.com/en-us/answers/questions/219889/bring-uwp-app-window-to-front-and-move-it.html

**Comment by manutalcual** [comment]
Unfortunately no. It could had be helpful a lot time ago, but now, no.
I've been investigating this a lot, a lot and a lot. There are so many distinct situations that having one app handling all well it's, not only almost impossible to someone like me (with not only no resources at all, but doing it at free time) but also almost impossible to anyone (including MS; ¡they didn't do it! why? they can't without breaking older apps.) 
To be able to handle this in an uniform way, all the apps should behave in the same way. That's not the case. And it will not be the case until a decade and several Windows versions pass through.

**Comment by stevennyman** [comment]
Ok, thank you for the response!

**Comment by manutalcual** [comment]
I would like to apologize. My response has been a bit hard. Sorry.
It is difficult to give a precise response. There are several adjustments, apps have to do to be able to, approximately, handle this: monitors (there may be more than one physical screen), desktops (virtual desktops I mean), resolution (DPI) and so on.
Each app handle that as they want and it is very difficult to move their windows, to reposition them, without breaking something. Moreover, some MS apps have an... "envelop" (a main, invisible window) that hides their windows (from a program perspective), so it is very, very difficult to handle that.
In addition to that, Windows lets apps handle how they want to handle all those things and doesn't give any (clever or easy) way to interfere with that. Their policy (as I understand from their docs) is: the app must have the right to "remember" window position, desktop and so on. (As some apps like Firefox do.) They even try to prevent apps to move another one's apps around. And that's completely right!
It is very easy to imagine someone, unrespectful developer moving windows around your desktop until you pay them. An "app position ransomware"!!
So I've given up trying to do more with this: as it is is enough for me (which was my only objective.)
I've shared the code and anyone can do whatever he want.

Again, I'm sorry for my previous response and also to not be able to do more with this.
And thanks for using it! I hope it is somewhat helpful for you anyway,.

**Comment by stevennyman** [comment]
Thanks for working on this! It seems like Windows 11 has added a [feature](https://old.reddit.com/r/Windows11/comments/qalzeh/do_the_remember_window_locations_based_on_monitor/) that is somewhat similar to what this program does, but since I'm still on Windows 10 I can't use it yet.
